state-sync/inter-component/consensus-notifications/src/lib.rs

```diff
@@ -0,0 +1,424 @@
+    pub(crate) callback: oneshot::Sender<ConsensusNotificationResponse>,
+}
+
+impl ConsensusSyncNotification {
+    pub fn new(
+        target: LedgerInfoWithSignatures,
+    ) -> (Self, oneshot::Receiver<ConsensusNotificationResponse>) {
+        let (callback, callback_receiver) = oneshot::channel();
+        let sync_notification = ConsensusSyncNotification { target, callback };
+
+        (sync_notification, callback_receiver)
+    }
+}
+
+/// The state sync component responsible for handling consensus requests and
+/// notifications.
+#[derive(Debug)]
+pub struct ConsensusNotificationListener {
+    pub notification_receiver: mpsc::UnboundedReceiver<ConsensusNotification>,
+}
+
+impl ConsensusNotificationListener {
+    pub fn new(notification_receiver: mpsc::UnboundedReceiver<ConsensusNotification>) -> Self {
+        ConsensusNotificationListener {
+            notification_receiver,
+        }
+    }
+
+    /// Respond to the commit notification previously sent by consensus.
+    pub async fn respond_to_commit_notification(
+        &mut self,
+        consensus_commit_notification: ConsensusCommitNotification,
+        result: Result<(), Error>,
+    ) -> Result<(), Error> {
+        consensus_commit_notification
+            .callback
+            .send(ConsensusNotificationResponse { result })
+            .map_err(|error| Error::UnexpectedErrorEncountered(format!("{:?}", error)))
+    }
+
+    /// Respond to the sync notification previously sent by consensus.
+    pub async fn respond_to_sync_notification(
+        &mut self,
+        consensus_sync_notification: ConsensusSyncNotification,
+        result: Result<(), Error>,
+    ) -> Result<(), Error> {
+        consensus_sync_notification
+            .callback
+            .send(ConsensusNotificationResponse { result })
+            .map_err(|error| Error::UnexpectedErrorEncountered(format!("{:?}", error)))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{ConsensusNotification, ConsensusNotificationSender, ConsensusNotifier, Error};
+    use diem_crypto::{ed25519::Ed25519PrivateKey, HashValue, PrivateKey, SigningKey, Uniform};
+    use diem_types::{
+        account_address::AccountAddress,
+        block_info::BlockInfo,
+        chain_id::ChainId,
+        contract_event::ContractEvent,
+        event::EventKey,
+        ledger_info::{LedgerInfo, LedgerInfoWithSignatures},
+        transaction::{RawTransaction, Script, SignedTransaction, Transaction, TransactionPayload},
+    };
+    use futures::executor::block_on;
+    use move_core_types::language_storage::TypeTag;
+    use std::{collections::BTreeMap, time::Duration};
+    use tokio::runtime::{Builder, Runtime};
+
+    #[test]
+    fn test_commit_state_sync_not_listening() {
+        // Create runtime and consensus notifier
+        let runtime = create_runtime();
+        let _enter = runtime.enter();
+        let (consensus_notifier, mut consensus_listener) = ConsensusNotifier::new(1000);
```

Comment:
I noticed you have `1000` a bunch of places for this.  Maybe consider a default timeout, or a constant for it.

Reply:
Yeah, I've used `1000` for test-only values; the real timeout is specified via the state sync config. Once we land this PR, I'm planning to follow up with some smaller cleanups to this code, so I can tackle a test-only constant then.